test/stdlib/DoubleWidth.swift

```diff
@@ -0,0 +1,345 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var dwTests = TestSuite("DoubleWidth")
+
+typealias UInt128 = DoubleWidth<UInt64>
+typealias UInt256 = DoubleWidth<UInt128>
+typealias UInt512 = DoubleWidth<UInt256>
+typealias UInt1024 = DoubleWidth<UInt512>
+
+typealias Int128 = DoubleWidth<Int64>
+typealias Int256 = DoubleWidth<Int128>
+typealias Int512 = DoubleWidth<Int256>
+typealias Int1024 = DoubleWidth<Int512>
+
+dwTests.test("Literals") {
+  let w: DoubleWidth<UInt8> = 100
+  expectTrue(w == 100 as Int)
+  
+  let x: DoubleWidth<UInt8> = 1000
+  expectTrue(x == 1000 as Int)
+  
+  let y: DoubleWidth<Int8> = 1000
+  expectTrue(y == 1000 as Int)
+  
+  let z: DoubleWidth<Int8> = -1000
+  expectTrue(z == -1000 as Int)
+  
+  expectCrashLater()
+  _ = -1 as DoubleWidth<UInt8>
+}
+
+dwTests.test("Literals/Large/Signed") {
+  let a: Int256 =
+    0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+  let b: Int256 =
+    -0x8000000000000000000000000000000000000000000000000000000000000000
+  expectEqual(a, Int256.max)
+  expectEqual(b, Int256.min)
+  expectCrashLater()
+  _ = -0x8000000000000000000000000000000000000000000000000000000000000001
+    as Int256
+}
+
+dwTests.test("Literals/Large/SignedOverflow") {
+  expectCrashLater()
+  _ = 0x8000000000000000000000000000000000000000000000000000000000000000
+    as Int256
+}
+
+dwTests.test("Literals/Large/Unsigned") {
+  let a: UInt256 =
+    0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+  let b: UInt256 = 0
+  expectEqual(a, UInt256.max)
+  expectEqual(b, UInt256.min)
+  expectCrashLater()
+  _ = -1 as UInt256
+}
+
+dwTests.test("Literals/Large/UnsignedOverflow") {
+  expectCrashLater()
+  _ = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0
+    as UInt256
+}
+
+dwTests.test("Arithmetic/unsigned") {
+  let x: DoubleWidth<UInt8> = 1000
+  let y: DoubleWidth<UInt8> = 1111
+  expectEqual(x + 1, 1001)
+  expectEqual(x + x, 2000)
+  expectEqual(x - (1 as DoubleWidth<UInt8>), 999)
```

Comment:
Is this `as` needed? Is there still a `Strideable` overload for `-` kicking int?

Reply:
Yep: `test/stdlib/DoubleWidth.swift:74:17: error: ambiguous use of operator '-'` along with a pile of warnings.

<details>
<summary>A pile of warnings</summary>
<pre><code>test/stdlib/DoubleWidth.swift:75:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:76:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(y - x, 111)
                ^
test/stdlib/DoubleWidth.swift:72:17: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x + 1, 1001)
                ^
test/stdlib/DoubleWidth.swift:75:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:76:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(y - x, 111)
                ^
test/stdlib/DoubleWidth.swift:93:17: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x + 1, 1001)
                ^
test/stdlib/DoubleWidth.swift:95:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - (1 as DoubleWidth<Int8>), 999)
                ^
test/stdlib/DoubleWidth.swift:96:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:97:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(0 - x, -1000)
                ^
test/stdlib/DoubleWidth.swift:99:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - y, 2111)
                ^
test/stdlib/DoubleWidth.swift:93:17: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x + 1, 1001)
                ^
test/stdlib/DoubleWidth.swift:95:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - (1 as DoubleWidth<Int8>), 999)
                ^
test/stdlib/DoubleWidth.swift:96:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:97:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(0 - x, -1000)
                ^
test/stdlib/DoubleWidth.swift:99:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - y, 2111)
                ^
test/stdlib/DoubleWidth.swift:166:38: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.min) - 1))
                                     ^
test/stdlib/DoubleWidth.swift:167:38: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.max) + 1))
                                     ^
test/stdlib/DoubleWidth.swift:169:26: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.min) - 1 < Int.min)
                         ^
test/stdlib/DoubleWidth.swift:170:26: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.max) + 1 > Int.max)
                         ^
test/stdlib/DoubleWidth.swift:166:38: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.min) - 1))
                                     ^
test/stdlib/DoubleWidth.swift:167:38: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.max) + 1))
                                     ^
test/stdlib/DoubleWidth.swift:169:26: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.min) - 1 < Int.min)
                         ^
test/stdlib/DoubleWidth.swift:170:26: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.max) + 1 > Int.max)
</code></pre>
</details>